src/python/grpcio_tests/tests/unit/framework/common/__init__.py

```diff
@@ -17,17 +17,27 @@
 import socket
 
 _DEFAULT_SOCK_OPTION = socket.SO_REUSEADDR if os.name == 'nt' else socket.SO_REUSEPORT
+_UNRECOVERABLE_ERRORS = ('Address already in use',)
+
+
+def _exception_is_unrecoverable(e):
+    for error in _UNRECOVERABLE_ERRORS:
+        if error in str(e):
+            return True
+    return False
 
 
 def get_socket(bind_address='localhost',
+               port=0,
                listen=True,
                sock_options=(_DEFAULT_SOCK_OPTION,)):
-    """Opens a socket bound to an arbitrary port.
+    """Opens a socket.
 
     Useful for reserving a port for a system-under-test.
 
     Args:
       bind_address: The host to which to bind.
+      port: The port to bind.
```

Comment:
Nit: s/The port to bind./The port to which to bind./

Reply:
Done.